File: app/src/main/java/cc/tomko/outify/core/AuthManager.java
package cc.tomko.outify.core;

public class AuthManager {
    /**
     * Checks for cached credentials.
     * Performs raw filesystem check
     */
    public native boolean hasCachedCredentials();

    /**
    * Processes the OAuth Code and caches the credentials into storage if success
     */
    public native boolean handleOAuthCode(String code, String state);

	/**
	 * Returns URL of Spotify authorization page
	 */
	public native String getAuthorizationURL();
}

File: app/src/main/java/cc/tomko/outify/core/AuthManager.kt
package cc.tomko.outify.core

import jakarta.inject.Inject
import jakarta.inject.Singleton

@Singleton
class AuthManager @Inject constructor() {
    /**
     * Checks for cached credentials.
     * Performs raw filesystem check
     */
    external fun hasCachedCredentials(): Boolean

    /**
     * Processes the OAuth Code and caches the credentials into storage if success.
     * Returns JSON: {"success":true} on success, {"error":{"type":"...","message":"..."}} on failure.
     */
    external fun handleOAuthCode(code: String, state: String? = ""): String

    /**
     * Returns URL of Spotify authorization page
     */
    external fun getAuthorizationURL(): String

    /**
     * Deletes the credentials cache file for Spirc
     */
    external fun logout(): Boolean
}
